nautobot/core/templates/generic/object_retrieve.html

```diff
@@ -154,6 +183,45 @@ <h1>
                 </div>
             </div>
         </div>
+        {% if associated_contacts_table %}
+            <div id="contacts" role="tabpanel" class="tab-pane {% if request.GET.tab == 'contacts' %}active{% else %}fade{% endif %}">
+                <div class="row">
+                    <div class="col-md-12">
+                        <form method="post">
+                            {% csrf_token %}
+                            <div class="panel panel-default">
+                                <div class="panel-heading">
+                                    <strong>Contact Associations</strong>
+                                    <div class="pull-right noprint">
+                                        {% if request.user.is_authenticated %}
+                                            <button type="button" class="btn btn-default btn-xs" data-toggle="modal" data-target="#AssociatedContactsTable_config" title="Configure table"><i class="mdi mdi-cog"></i> Configure</button>
+                                        {% endif %}
+                                    </div>
+                                </div>
+                                {% render_table associated_contacts_table 'inc/table.html' %}
+                                <div class="panel-footer noprint">
+                                    {% if perms.dcim.delete_contactassociation %}
+                                        <button type="submit" formaction="{% url 'extras:contactassociation_bulk_delete' %}?return_url={{request.path}}" class="btn btn-danger btn-xs">
+                                            <span class="mdi mdi-trash-can-outline" aria-hidden="true"></span> Delete
+                                        </button>
+                                    {% endif %}
+                                    {% if perms.dcim.add_contactassociation %}
```

Comment:
Certainly we need to at least be self-consistent about permissions that we're checking :-)